sinopia_client/test/api/LDPApi.spec.js

```diff
@@ -54,29 +54,71 @@
   const chai = require('chai')
   expect = chai.expect // no 'var', expect is already a param
 
+  const fs = require('fs');
+  const path = require('path');
+
   describe('LDPApi', function() {
     describe('createGroup', function() {
+      // need base container to exist, since group is created under that
+      beforeEach(function() {
+        let rsrcCtx = new SinopiaServer.SinopiaBasicContainerContext('http://www.w3.org/2000/01/rdf-schema#', 'http://www.w3.org/ns/ldp#')
+        let baseRsrc = new SinopiaServer.SinopiaBasicContainer('', rsrcCtx, ['ldp:Container', 'ldp:BasicContainer'], 'Sinopia LDP Server')
+        return instance.updateBase(baseRsrc).catch(function(err) { console.error(`Error setting up base container: ${err}`) })
+      });
+
       it('should call createGroup successfully', function() {
-        // created manually for now, but will need to do this as setup here (and teardown later?) to make stuff created under /repository not 404
-        // instance.updateBase(new SinopiaServer.Resource( , , new SinopiaServer.ResourceContext));
-        // var resources = [new SinopiaServer.Resource()];
-        // var group = new SinopiaServer.LDPContainer('', 'PCC Group', null, resources);
-        // return instance.createGroup('pcc', group, { contentType: 'application/ld+json' })
-        //   .then(function(_data) {
-        //     expect().to.be();
-        //   });
+        let rand_num = Math.floor(Math.random() * 100)
+        let rsrcCtx = new SinopiaServer.SinopiaBasicContainerContext('http://www.w3.org/2000/01/rdf-schema#', 'http://www.w3.org/ns/ldp#')
+        let groupRsrc = new SinopiaServer.SinopiaBasicContainer('', rsrcCtx, ['ldp:Container', 'ldp:BasicContainer'], 'PCC Group' + rand_num)
+
+        return instance.createGroup('pcc', groupRsrc).then(function(_responseData) {
+          return instance.getGroup('pcc')
+            .then(function(responseData) {
+              expect(responseData['@id']).to.equal('http://localhost:8080/repository/pcc')
+              expect(responseData['@context']).to.deep.equal(groupRsrc['@context'])
+              expect(responseData['label']).to.equal(groupRsrc['rdfs:label'])
+            })
+        })
+        //TODO: attempting creation again should result in a response of HTTP 409 conflict
```

Comment:
I always like issue numbers attached to TODOs in code ...

Reply:
will file issues/update comments